### ISSUE TITLE
False positive on UnusedImport with const variable import

### ISSUE THREAD

**Issue by dimsuz** [detekt/detekt#4689]
The following code when run with type resolution is reporting an error:

```kotlin
import com.example.BuildConfig.FILE_PROVIDER_AUTHORITY
import com.example.BuildConfig.FLAVOR

class ApplicationDelegate {
  override fun getFlavor(): String = FLAVOR
  override fun getFileProviderAuthority(): String = FILE_PROVIDER_AUTHORITY
}
```
The file `BuildConfig` is a java file, generated by the Android Gradle Plugin:
```java
public final class BuildConfig {
  public static final String FLAVOR = "myflavor";
  public static final String FILE_PROVIDER_AUTHORITY = "com.example.debug.provider";
}
```

The error is:

> * The import 'com.example.BuildConfig.FILE_PROVIDER_AUTHORITY' is unused. [UnusedImports]
> * The import 'com.example.BuildConfig.FLAVOR' is unused. [UnusedImports]

## Your Environment
* Version of detekt used: `1.20.0-RC2`
* Version of Gradle used (if applicable): `7.4.x`

**Comment by eygraber** [comment]
@BraisGabin dupe of https://github.com/detekt/detekt/issues/4676?

**Comment by BraisGabin** [comment]
I wouldn't consider it a dope but the issue is probably the same... I'm wondering what created all this new issues... It seems like a change in the compiler behavior
